lib/semantics/resolve-names.cpp

```diff
@@ -6076,6 +6095,53 @@ bool OmpAttributeVisitor::Pre(const parser::OpenMPLoopConstruct &x) {
   return true;
 }
 
+const parser::Name &OmpAttributeVisitor::GetLoopIndex(
+    const parser::DoConstruct &x) {
+  auto &loopControl{x.GetLoopControl().value()};
+  using Bounds = parser::LoopControl::Bounds;
+  const Bounds &bounds{std::get<Bounds>(loopControl.u)};
+  return bounds.name.thing;
+}
+
+void OmpAttributeVisitor::ResolveSeqLoopIndexInParallelOrTaskConstruct(
+    const parser::Name &iv) {
+  Symbol *symbol{nullptr};
+  std::vector<OmpContext>::reverse_iterator targetIt;
+  for (auto it{ompContext_.rbegin()}; it != ompContext_.rend(); ++it) {
+    if (parallelSet.test(it->directive) ||
+        taskGeneratingSet.test(it->directive)) {
+      symbol = ResolveOmp(iv, Symbol::Flag::OmpPrivate, it->scope);
+      targetIt = it;
+      break;
+    }
+  }
+  if (symbol) {
```

Comment:
I think this is a better way to write this:
```
  auto targetIt{ompContext_.rbegin()};
  for (;; ++targetIt) {
    if (targetIt == ompContext_.rend()) {
      return;
    }
    if (parallelSet.test(targetIt->directive) ||
        taskGeneratingSet.test(targetIt->directive)) {
      break;
    }
  }
  if (Symbol *symbol{ResolveOmp(iv, Symbol::Flag::OmpPrivate, targetIt->scope)}) {
```

Reply:
thank you for the detailed suggestion! I've made the changes.